lms/djangoapps/courseware/access_utils.py

```diff
@@ -78,4 +79,5 @@ def in_preview_mode():
     Returns whether the user is in preview mode or not.
     """
     hostname = get_current_request_hostname()
-    return bool(hostname and settings.PREVIEW_DOMAIN in hostname.split('.'))
+    preview_lms_base = settings.FEATURES.get('PREVIEW_LMS_BASE', None)
+    return bool(preview_lms_base and hostname and re.match(preview_lms_base.split(':')[0], hostname))
```

Comment:
Why are we using `re.match` here instead of doing a simple string comparison like `hostname in preview_lms_base.spilt(':')`?